src/utils/get-config.test.ts

```diff
@@ -41,6 +41,22 @@ describe('#getConfig(true)', () => {
 		});
 	});
 
+	it('Handles compile errors in JS config files', async () => {
+		jest.spyOn(process, 'cwd').mockReturnValue(path.join(testFileTreesPath, 'js-config-compile-error'));
+
+		await expect(() => getConfig(true)).rejects.toThrow(
+			'An error was encountered while trying to compile rugged.config.js (see below):\n\n'
+		);
+	});
+
+	it('Handles compile errors in TS config files', async () => {
+		jest.spyOn(process, 'cwd').mockReturnValue(path.join(testFileTreesPath, 'ts-config-compile-error'));
+
+		await expect(() => getConfig(true)).rejects.toThrow(
```

Comment:
Usually we test what type of error it throws, too, with a line above this like:

```ts
await expect(() => getConfig(true)).rejects.toThrow(PrintableError);
```

Reply:
I actually did this at first but I was having an issue where on the second test it would compile the custom config to an empty object instead of throwing an error, and I wanted to ask you if you had any ideas about why it might be doing that

Reply:
I can try to investigate; I'll check out this branch and do some testing...

Reply:
Okay, it looks if you `require()` a file more than once, and if it had a compile error the first time, the `require()` function just hands back an empty object instead of throwing... weird.

Reply:
I pushed up a fix; basically, we have to capture the error once, then run multiple tests against it after.